packages/wonder-blocks-form/src/components/labeled-text-field.js

```diff
@@ -1,21 +1,56 @@
 // @flow
 import * as React from "react";
 
+import {IDProvider} from "@khanacademy/wonder-blocks-core";
+import {type Typography} from "@khanacademy/wonder-blocks-typography";
+
 import FieldHeading from "./field-heading.js";
-import TextField from "./text-field.js";
+import TextField, {type TextFieldType} from "./text-field.js";
 
 type Props = {|
+    /**
+     * An optional unique identifier for the TextField.
+     * If no id is specified, a unique id will be auto-generated.
+     */
+    id?: string,
+
+    /**
+     * Determines the type of input. Defaults to text.
+     */
+    type: TextFieldType,
+
+    /**
+     * Provide a label for the TextField.
+     */
+    label: string | React.Element<Typography>,
+
+    /**
+     * Provide a description for the TextField.
+     */
+    description?: string | React.Element<Typography>,
+
+    /**
+     * The initial input value.
+     */
+    initialValue?: string,
+
     /**
      * Makes a read-only input field that cannot be focused. Defaults to false.
      */
     disabled: boolean,
 |};
 
 type DefaultProps = {|
+    type: $PropertyType<Props, "type">,
     disabled: $PropertyType<Props, "disabled">,
 |};
 
 type State = {|
+    /**
+     * The value of the input.
+     */
+    value: string,
```

Comment:
This is another area that I forgot to add in the implementation spec. I have added `value` as a state (will update implementation spec).